react-state-3/react-state-3_adding-animals/src/App.jsx

```diff
@@ -1,31 +1,35 @@
-import { useState } from "react";
-import Form from "./components/Form";
-import List from "./components/List";
-import "./App.css";
+import { useState } from 'react';
+import { uid } from 'uid';
 
+import Form from './components/Form';
+import List from './components/List';
+
+import './App.css';
 const initialAnimals = [
   {
-    id: "mTBErig",
-    name: "Octopus",
-    emoji: "🐙",
+    id: 'mTBErig',
+    name: 'Octopus',
+    emoji: '🐙',
   },
   {
-    id: "iMsbXQ1",
-    name: "Crocodile",
-    emoji: "🐊",
+    id: 'iMsbXQ1',
+    name: 'Crocodile',
+    emoji: '🐊',
   },
   {
-    id: "Qjf_K7V",
-    name: "Beaver",
-    emoji: "🦫",
+    id: 'Qjf_K7V',
+    name: 'Beaver',
+    emoji: '🦫',
   },
 ];
 
 export default function App() {
   const [animals, setAnimals] = useState(initialAnimals);
 
   function handleAddAnimal(newAnimal) {
-    console.log(newAnimal);
+    const newArray = [...animals, { ...newAnimal, id: uid() }];
+    setAnimals(newArray);
```

Comment:
another option would be: 
setAnimals([...animals, { ...newAnimal, id: uid() }]);
but i guess you know that ;-)